### ISSUE TITLE
Filter is silly on numbers

### ISSUE THREAD

**Issue by paramecie** [rgriebl/brickstore#594]
2022.4.2 (Legacy)

* Happens to any list
* For this test import inventory Set 3333
* Click Filter then type 9999 in the filter text field
* It will display 8 lots on 13, where obviously nothing corresponds to "9999"

I've found out it corresponds to the Weight column.
It gives the same list for Filter: Weight/Contains/9999

**Comment by rgriebl** [comment]
This bug is actually in BrickStore since 2021.3.1 (over 1,5 years). What happens is that the weight is not rounded to 3 decimals before comparing, so BrickStore does
`"1.3799999952316284".contains("9999")` which is obviously true :)

**Comment by rgriebl** [comment]
The main problem here is that some filters (contains/is) should act on the visible string and others (greater/less) should act on the actual numeric value.
Now some columns have no numeric value, while the visible string of the numeric columns is locale dependent and computed a few layers above the filter code.
TL;DR: the filter code has always been a mess :)

**Comment by rgriebl** [comment]
One more thing: if you have a price of `1000`, what should a filter `contains(1000)` do?
- without any localization the number would be `"1000"`
- in English it is `"1,000"`
- in German it is `"1.000"`
- in Spanish it is `"1000"` (yes those guys are weird: #555)

So should it match regardless of current locale or not? Both yes and no are valid answers

**Comment by paramecie** [comment]
I know it's old, I've seen it for a long time but I didn't complain :-)

About 1000, we should search on 1000, independently of the localization.
And searching 1234 would find 1,234.00

On the other hand searching for "1,234" (in English) should NOT find 1,234 (to be consistent)?

**Comment by rgriebl** [comment]
Just noticed: the fun continues with imperial weights! I forgot to take that into account, so the filter acts on grams, while the display is in oz/lb. Nobody ever complained about that, so I guess the guys using banana measurements never filter on weight :)
